packages/walletconnect-solana/src/adapter.ts

```diff
@@ -115,16 +117,52 @@ export class WalletConnectWallet {
         }
     }
 
-    async signTransaction(transaction: Transaction): Promise<Transaction> {
+    async signTransaction<T extends Transaction | VersionedTransaction>(transaction: T): Promise<T> {
         if (this._client && this._session) {
-            const { signature } = await this._client.request<{ signature: string }>({
-                chainId: this._network,
-                topic: this._session.topic,
-                request: { method: WalletConnectRPCMethods.signTransaction, params: { ...transaction } },
-            });
-            transaction.addSignature(this.publicKey, Buffer.from(base58.decode(signature)));
-
-            return transaction;
+            if (isVersionedTransaction(transaction)) {
+                const rawTransaction = Buffer.from(transaction.serialize()).toString('base64');
+                if (transaction.version === 'legacy') {
+                    const legacyTransaction = Transaction.from(transaction.serialize());
+                    const { signature } = await this._client.request<{ signature: string }>({
+                        chainId: this._network,
+                        topic: this._session.topic,
+                        request: {
+                            method: WalletConnectRPCMethods.signTransaction,
+                            params: { ...legacyTransaction, transaction: rawTransaction },
+                        },
+                    });
+                    transaction.addSignature(this.publicKey, base58.decode(signature));
+                    return transaction;
+                } else {
+                    const { signature } = await this._client.request<{ signature: string }>({
+                        chainId: this._network,
+                        topic: this._session.topic,
+                        request: {
+                            method: WalletConnectRPCMethods.signTransaction,
+                            params: { transaction: rawTransaction },
+                        },
```

Comment:
this whole block looks identical to the same request for the `legacy` transaction, i'd suggest combining the two blocks and just create the parameters differently depending on the incoming transaction